processor/transformprocessor/internal/traces/processor.go

```diff
@@ -19,26 +19,18 @@ import (
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/model/pdata"
-	"go.uber.org/multierr"
 	"go.uber.org/zap"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
 )
 
 type Processor struct {
-	queries []Query
+	queries []common.Query
 	logger  *zap.Logger
 }
 
-// Query holds a top level Query for processing trace data. A Query is a combination of a function
-// invocation and the condition to match telemetry for invoking the function.
-type Query struct {
-	function  exprFunc
-	condition condFunc
-}
-
 func NewProcessor(statements []string, functions map[string]interface{}, settings component.ProcessorCreateSettings) (*Processor, error) {
-	queries, err := Parse(statements, functions)
+	queries, err := common.ParseQueries(statements, functions, ParsePath)
```

Comment:
@Aneurysm9 For https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/8972#discussion_r842235790

There is some coupling here that `traces.ParsePath` is passed here, and then the function is applied to a `spanTransformContext` below. With generics, `queries` could be a parameterized type, until then not sure if there is anything better to do here but let me know any ideas.